### ISSUE TITLE
Add method to retrieve all NavigationProperyBindings.

### ISSUE THREAD

**Issue by chrisspre** [OData/odata.net#2060]
For deep insert scenarios and potentially others it is important for a service to inspect all navigation property bindings.
There is currently no convenient method that allows to do that.

Suggestion is to add a method to IEdmEntityContainer (probably an extension method) that effectively returns all NavigationProperyBindings

```csharp
public static IEnumerable<(IEdmEntityContainerElement, IEdmNavigationPropertyBinding)> FindBindings(this IEdmEntityContainer container);
```

There might be better representations of the return type. For example a collection of key/value pairs with the binding paths (an IEdmPathExpression?)  as the key and the IEdmNavigationPropertyBinding as the value. 

### Assemblies affected

"Microsoft.OData.Edm" Version="7.8.3"

add.*

**Comment by xuzhg** [comment]
@chrisspre IEdmNavigationSource interface has the property that retrieves all NPB for that navigation source. See https://github.com/OData/odata.net/blob/master/src/Microsoft.OData.Edm/Schema/Interfaces/IEdmNavigationSource.cs#L50

If you want to return all NPB in an entity set container, you can enumerate the navigation source (entityset & singleton) and returns the NPB for each navigation source.

**Comment by chrisspre** [comment]
Correct. This is possible to implement . But it is not obvious code. 

One needs to iterate over all Container elements 
```
foreach(var element in model.EntityContainer.Elements)
```
And then these elements need to be tested if they are EntitySet or Singleton.
There is as far as I know there is no interface that provides NavigationPropertyBindings  property for these two to code woudl look like:
```
                switch (element)
                {
                    case IEdmSingleton singleton:
                         // do something with singleton.NavigationPropertyBindings  ;
                    case IEdmEntitySet entitySet:
                         // do something with entitySet.NavigationPropertyBindings 
                    default:
                        continue;
                }
```

I would like to see this coming with the framework, in a productized. maintained, tested way .

**Comment by KenitoInc** [comment]
Resolved by #2157
